streamwebs_frontend/streamwebs/models.py

```diff
@@ -233,6 +232,73 @@ class Meta:
         verbose_name_plural = 'Water Quality'
 
 
+class CameraPointManager(models.Manager):
+
+    def create_camera_point(self, site, cp_date, created_by='', latitude=None,
+                            longitude=None, map_datum='', description=''):
+
+        return self.create(site=site, cp_date=cp_date, created_by=created_by,
+                           latitude=latitude, longitude=longitude,
+                           map_datum=map_datum, description=description)
+
+
+class CameraPoint(models.Model):
+    site = models.ForeignKey(Site, on_delete=models.CASCADE, null=True)
+    cp_date = models.DateField(default=datetime.date.today)
+    created_by = models.CharField(max_length=255, blank=True)
+    latitude = models.DecimalField(default=0, max_digits=9, decimal_places=6,
+                                   blank=True, null=True)
+    longitude = models.DecimalField(default=0, max_digits=9, decimal_places=6,
+                                    blank=True, null=True)
+    map_datum = models.CharField(max_length=255, blank=True)
+    description = models.TextField(blank=True)
+
+    camera_points = CameraPointManager()
+
+    def __str__(self):
+        return ('Camera point ' + str(self.id) + ' for site ' +
+                self.site.site_name)
+
+    class Meta:
+        verbose_name = 'Camera Point'
+        verbose_name_plural = 'Camera Points'
+
+
+class PhotoPointManager(models.Manager):
+
+    def create_photo_point(self, camera_point, pp_date, compass_bearing,
+                           distance, camera_height, photo_filename='',
+                           photo=None, notes=''):
+
+        return self.create(camera_point=camera_point, pp_date=pp_date,
+                           compass_bearing=compass_bearing, distance=distance,
+                           camera_height=camera_height,
+                           photo_filename=photo_filename, photo=photo,
+                           notes=notes)
+
+
+class PhotoPoint(models.Model):
+    camera_point = models.ForeignKey(CameraPoint, on_delete=models.CASCADE,
+                                     null=True, related_name='camera_point')
+    pp_date = models.DateField(default=datetime.date.today)
+    compass_bearing = models.PositiveSmallIntegerField()
+    distance = models.DecimalField(max_digits=3, decimal_places=0)
+    camera_height = models.DecimalField(max_digits=3, decimal_places=0)
+    photo_filename = models.CharField(max_length=255, blank=True)
```

Comment:
Do you need the filename when you have ImageField? You should be able to get the filename from the imagefield